servers/src/common/adapters.rs

```diff
@@ -134,8 +134,8 @@ impl p2p::ChainAdapter for NetToChainAdapter {
 				.upgrade()
 				.expect("failed to upgrade weak ref to chain");
 
-			if let Ok(prev_header) = chain.get_block_header(&cb.header.previous) {
-				if let Ok(()) = block.validate(&prev_header) {
+			if let Ok(prev_block) = chain.get_block(&cb.header.previous) {
+				if let Ok(()) = block.validate(&prev_block.header) {
```

Comment:
I think that means anyone could send us an invalid compact blocks to force us to load full blocks from the db, a relatively expensive operation. Why not completely refusing compact blocks during sync? We have no use for them anyway.